main.py

```diff
@@ -68,27 +67,50 @@ def do_create_room(self, arg):
     def do_add_person(self,arg):
         """Usage: add_person <person_fname> <person_lname> <FELLOW/STAFF> [<wants_accommodation>]"""
         check = Dojo().add_person(arg)
-        if check == True:
-                Dojo().assign_room(arg)
+        if check:
+                Dojo().allocate_room(arg)
```

Comment:
Done. 
I have implemented the singleton pattern for instantiating my class.
Thanks.